frontend/luadata.lua

```diff
@@ -0,0 +1,92 @@
+--[[--
+Handles append-mostly data such as KOReader's bookmarks and dictionary search history.
+]]
+
+local LuaSettings = require("luasettings")
+local dump = require("dump")
+local logger = require("logger")
+local util = require("util")
+
+local LuaData = LuaSettings:new{
+    name = nil,
+}
+--- Opens a LuaData file.
+function LuaData:open(file_path, name)
+    if name then self.name = name end
+    local new = {file=file_path, data={}}
+
+    -- some magic to allow for self-describing function names
+    local _local = {}
+    _local.__index = _local
+    setmetatable(_G, _local)
```

Comment:
Is that _polluting_ the global namespace with LookupHistoryEntry and such? If yes, any way to revert it when no more needed?
Not that I really care :) we're making sure everywhere things are local, so it shouldn't hurt.

Reply:
It's `local _local` so it doesn't. ;-)

```lua
-- bla.lua
do
	local _local = {}
	_local.__index = _local
	setmetatable(_G, _local)
	_local["Entry"] = "whatever"
	print(Entry)
end

_local["AnotherEntry"] = "bla"
```

```lua
> dofile("bla.lua")
whatever
bla.lua:10: attempt to index global '_local' (a nil value)
stack traceback:
	bla.lua:10: in main chunk
	[C]: in function 'dofile'
	stdin:1: in main chunk
	[C]: in ?
```

Reply:
ok, so that was indeed more magical that I thought :)